test-suite-utils/src/main/groovy/io/micronaut/security/testutils/Keycloak.groovy

```diff
@@ -51,21 +53,22 @@ class Keycloak {
 
     static void init() {
         if (keycloak == null) {
-            keycloak = new GenericContainer("jboss/keycloak:8.0.0")
-                    .withExposedPorts(8080)
+            if (OperatingSystem.current.macOs && System.getProperty("os.arch") == 'aarch64') {
+                keycloak = new GenericContainer(new ImageFromDockerfile("keycloak-m1", false).withFileFromClasspath("Dockerfile", "/Dockerfile.keycloak"))
+            } else {
+                keycloak = new GenericContainer("jboss/keycloak:16.1.1")
+            }
+
+            keycloak = keycloak.withExposedPorts(8080)
                     .withEnv([
                             KEYCLOAK_USER: 'user',
                             KEYCLOAK_PASSWORD: 'password',
                             DB_VENDOR: 'H2',
                     ])
-                    .waitingFor(new LogMessageWaitStrategy().withRegEx(".*Deployed \"keycloak-server.war\".*").withStartupTimeout(Duration.ofMinutes(2)))
+                    .waitingFor(new LogMessageWaitStrategy().withRegEx(".*Deployed \"keycloak-server.war\".*").withStartupTimeout(Duration.ofMinutes(5)))
             keycloak.start()
             keycloak.execInContainer("/opt/jboss/keycloak/bin/kcreg.sh config credentials --server http://localhost:8080/auth --realm master --user user --password password".split(" "))
-            keycloak.execInContainer("/opt/jboss/keycloak/bin/kcreg.sh create -s clientId=$CLIENT_ID -s redirectUris=[\"http://${getRedirectUriHost()}*\"]".split(" "))
-            Container.ExecResult result = keycloak.execInContainer("/opt/jboss/keycloak/bin/kcreg.sh get $CLIENT_ID".split(" "))
-            Map map = new ObjectMapper()
-                    .readValue(result.getStdout(), Map.class)
-            clientSecret = map.get("secret")
+            keycloak.execInContainer("/opt/jboss/keycloak/bin/kcreg.sh create -s clientId=$CLIENT_ID -s redirectUris=[\"http://${getRedirectUriHost()}*\"] -s secret=$clientSecret".split(" "))
```

Comment:
Is the idea to no longer get a clientsecret from keycloak but provide one istead?

Reply:
Keycloak no longer auto-generates a client secret, this was the easiest way to get one set up